hunts/chart_utils.py

```diff
@@ -0,0 +1,69 @@
+from django.utils import timezone
+
+from .models import Hunt
+from puzzles.models import Puzzle
+
+
+def can_use_chart(hunt):
+    if not hunt.start_time:
+        return hunt.get_num_unlocked() > 0
+
+    return timezone.now() > hunt.start_time
+
+
+# Returns a tuple of lists of solved puzzle data, sorted by puzzle solve time.
+# labels: Names of solved puzzles
+# times: Solve times of solved puzzles, in ISO 8601 format strings
+# counts: Total solves/unlocks at the time of that puzzle's solve/unlock (inclusive)
+# is_meta: Whether the solved puzzle is meta (list of booleans)
+def get_chart_data(hunt, unlocks=False):
+
+    # if start_time not given, use first puzzle creation time
+    # (use can_use_chart() to check beforehand)
+    chart_start_time = hunt.start_time
+    if not chart_start_time:
+        chart_start_time = hunt.puzzles.earliest("created_on").created_on
```

Comment:
Can't .earliest return None? Does this throw an error if there are no puzzles in the hunt? I guess the idea is you should call `can_use_chart` first before calling this function but maybe there should be some handling at the start of this function too

Reply:
Yeah, the idea was that `can_use_chart` would always be called before `get_chart_data`, but some handling within the function would also be an option. If we were to do that, I'd probably get rid of `can_use_chart` altogether and reincorporate it into the body of `get_chart_data`.
(Actually, that does seem more sensible, now that I think about it.)